includes/discovery/ntp/awplus.inc.php

```diff
@@ -0,0 +1,130 @@
+        $result['port'] = '123'; // awplus only supports default NTP Port.
+        $result['stratum'] = $atNtpAssociationEntry[$index]['atNtpAssociationStratum'];
+        $result['peerref'] = $atNtpAssociationEntry[$index]['atNtpAssociationRefClkAddr'];
+        $result['label'] = $result['peer'].":".$result['port'];
+
+        // Set the status, 16 = Bad
+        if ($result['stratum'] == 16) {
+            $result['status'] = 2;
+            $result['error'] = 'NTP is not in sync';
+        } else {
+            $result['status'] = 0;
+            $result['error'] = '';
+        }
+
+        d_echo("NTP Peer found: ");
+        d_echo($result);
+        $tblComponents[] = $result;
+    }
+
+    /*
+     * Ok, we have our 2 array's (Components and SNMP) now we need
+     * to compare and see what needs to be added/updated.
+     *
+     * Let's loop over the SNMP data to see if we need to ADD or UPDATE any components.
+     */
+    foreach ($tblComponents as $key => $array) {
+        $component_key = false;
+
+        // Loop over our components to determine if the component exists, or we need to add it.
+        foreach ($components as $compid => $child) {
+            if ($child['UID'] === $array['UID']) {
+                $component_key = $compid;
+            }
+        }
+
+        if (!$component_key) {
+            // The component doesn't exist, we need to ADD it - ADD.
+            $new_component = $component->createComponent($device['device_id'], $module);
+            $component_key = key($new_component);
+            $components[$component_key] = array_merge($new_component[$component_key], $array);
+            echo "+";
+        } else {
+            // The component does exist, merge the details in - UPDATE.
+            $components[$component_key] = array_merge($components[$component_key], $array);
+            echo ".";
+        }
+    }
+
+    /*
+     * Loop over the Component data to see if we need to DELETE any components.
+     */
+    foreach ($components as $key => $array) {
+        // Guilty until proven innocent
+        $found = false;
+
+        foreach ($tblComponents as $k => $v) {
+            if ($array['UID'] == $v['UID']) {
+                // Yay, we found it...
+                $found = true;
+            }
+        }
+
+        if ($found === false) {
+            // The component has not been found. we should delete it.
+            echo "-";
+            $component->deleteComponent($key);
+        }
+    }
+
+    // Write the Components back to the DB.
+    $component->setComponentPrefs($device['device_id'], $components);
+    echo "\n";
+} // End if not error
+
+$module = strtolower($module);
+if (count($components) > 0) {
+    if (dbFetchCell('SELECT COUNT(*) FROM `applications` WHERE `device_id` = ? AND `app_type` = ?', array($device['device_id'], $module)) == '0') {
+        dbInsert(array('device_id' => $device['device_id'], 'app_type' => $module), 'applications');
```

Comment:
When creating the app in the database, you need to supply an empty string (or valid data) for the app_status field.

Reply:
Thanks @murrant. I see now. 

So this works, and I assume is the correct way to do it:

```
+        dbInsert(array('device_id' => $device['device_id'], 'app_type' => $module, 'app_status' => '', 'app_instance' => ''), 'applications');
```

I guess I should fix up the same line in the cisco.inc.php config as well ? 

https://github.com/mattie47/librenms/blob/3728439fd88ffafb395f60c4202da11d151612d1/includes/discovery/ntp/cisco.inc.php#L126